src/metacognition/failure_predictor.py

```
@@ -0,0 +1,451 @@
+                elif usage_ratio >= 0.5:
+                    return 0.2
+
+        # Check context for rate limit hints
+        if context.get("rate_limit_warning"):
+            return 0.7
+
+        return 0.0
+
+    def _analyze_context(
+        self,
+        action: str,
```

Ruff (0.14.10)

Unused method argument: `action`

(ARG002)